pyomo/contrib/solver/tests/solvers/test_ipopt.py

```diff
@@ -28,6 +34,248 @@
 """
 
 
+ipopt_available = ipopt.Ipopt().available()
+
+
+@unittest.skipIf(not ipopt_available, "The 'ipopt' command is not available")
+class TestIpoptSolverConfig(unittest.TestCase):
+    def test_default_instantiation(self):
+        config = ipopt.IpoptConfig()
+        # Should be inherited
+        self.assertIsNone(config._description)
+        self.assertEqual(config._visibility, 0)
+        self.assertFalse(config.tee)
+        self.assertTrue(config.load_solutions)
+        self.assertTrue(config.raise_exception_on_nonoptimal_result)
+        self.assertFalse(config.symbolic_solver_labels)
+        self.assertIsNone(config.timer)
+        self.assertIsNone(config.threads)
+        self.assertIsNone(config.time_limit)
+        # Unique to this object
+        self.assertIsInstance(config.executable, type(Executable('path')))
+        self.assertIsInstance(config.writer_config, type(NLWriter.CONFIG()))
+
+    def test_custom_instantiation(self):
+        config = ipopt.IpoptConfig(description="A description")
+        config.tee = True
+        self.assertTrue(config.tee)
+        self.assertEqual(config._description, "A description")
+        self.assertIsNone(config.time_limit)
+        # Default should be `ipopt`
+        self.assertIsNotNone(str(config.executable))
+        self.assertIn('ipopt', str(config.executable))
+        # Set to a totally bogus path
+        config.executable = Executable('/bogus/path')
+        self.assertIsNone(config.executable.executable)
+        self.assertFalse(config.executable.available())
+
+
+class TestIpoptSolutionLoader(unittest.TestCase):
+    def test_get_reduced_costs_error(self):
+        loader = ipopt.IpoptSolutionLoader(None, None)
+        with self.assertRaises(NoSolutionError):
+            loader.get_reduced_costs()
+
+        # Set _nl_info to something completely bogus but is not None
+        class NLInfo:
+            pass
+
+        loader._nl_info = NLInfo()
+        loader._nl_info.eliminated_vars = [1, 2, 3]
+        with self.assertRaises(NotImplementedError):
+            loader.get_reduced_costs()
```

Comment:
Can you leave a note that this test could go away some day if we enable returning duals when presolve is on?